### PR TITLE
[new release] gsl (1.25.0)

### PR DESCRIPTION
**Pull request by mmottl**
GSL - Bindings to the GNU Scientific Library

- Project page: <a href="https://mmottl.github.io/gsl-ocaml">https://mmottl.github.io/gsl-ocaml</a>
- Documentation: <a href="https://mmottl.github.io/gsl-ocaml/api">https://mmottl.github.io/gsl-ocaml/api</a>

##### CHANGES:

  * Make the GSL bindings compatible with OCaml 5.0.

    Thanks to Florian Angeletti <florian.angeletti@inria.fr> for this
    large patch!

  * Added `Randist.multinomial_inplace`

    Thanks to Martynas Sinkievic <tasmartynas@gmail.com> for this contribution!

### CODE REVIEW

**Comment by mmottl** [comment]
Not sure, but it seems `opam lint` is outdated somewhere. If I call it on my machine, it accepts the license without problems.

File: packages/gsl/gsl.1.25.0/opam [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+gsl-ocaml interfaces the GSL (GNU Scientific Library), providing many of the
+most frequently used functions for scientific computation including algorithms
+for optimization, differential equations, statistics, random number generation,
+linear algebra, etc."""
+maintainer: ["Markus Mottl <markus.mottl@gmail.com>"]
+authors: [
+  "Olivier Andrieu <oandrieu@gmail.com>"
+  "Markus Mottl <markus.mottl@gmail.com>"
+]
+license: "GPL-3.0+"
```
Comment (mseri) [inline packages/gsl/gsl.1.25.0/opam:13]:
```suggestion
license: "GPL-3.0-or-later"
```

File: packages/gsl/gsl.1.25.0/opam [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    "-p"
+    name
+    "-j"
+    jobs
+    "@install"
+    "@runtest" {with-test}
+    "@doc" {with-doc}
+  ]
+]
+dev-repo: "git+https://github.com/mmottl/gsl-ocaml.git"
```
Comment (mseri) [inline packages/gsl/gsl.1.25.0/opam:39]:
```suggestion
available: arch != "arm32"
dev-repo: "git+https://github.com/mmottl/gsl-ocaml.git"
```

**Comment by mmottl** [comment]
I've fixed the license specification as suggested here upstream. Not sure why `opam lint` did not report the problem on my local machine.

**Comment by mseri** [comment]
The CI here uses opam 2.2, I think it may have become stricter

**Comment by mseri** [comment]
Thanks

**Pull request merged**